ts/screens/authentication/cie/CieWrongCiePinScreen.tsx

```diff
@@ -1,94 +1,187 @@
+      testID: `message-action-${label}`
+    }),
+    []
+  );
+
+  const messages: Messages = React.useMemo(
+    () => ({
+      2: {
+        pictogram: "attention",
+        title: I18n.t("authentication.cie.pin.incorrectCiePinTitle1"),
+        subtitle: I18n.t("authentication.cie.pin.incorrectCiePinContent1"),
+        action: createMessageAction({
+          label: I18n.t("global.buttons.retry"),
+          onPress: navigateToCiePinScreen
+        }),
+        secondaryAction: createMessageAction({
+          label: I18n.t("global.buttons.close"),
+          onPress: navigateToAuthenticationScreen
+        })
+      },
+      1: {
+        pictogram: "attention",
+        title: I18n.t("authentication.cie.pin.incorrectCiePinTitle2"),
+        subtitle: I18n.t("authentication.cie.pin.incorrectCiePinContent2"),
+        action: createMessageAction({
+          label: I18n.t("global.buttons.retry"),
+          onPress: navigateToCiePinScreen
+        }),
+        secondaryAction: createMessageAction({
+          label: I18n.t(
+            "authentication.cie.pin.incorrectCiePinSecondaryActionLabel2"
+          ),
+          onPress: didYouForgetPin
+        })
+      },
+      0: {
+        pictogram: "fatalError",
+        title: I18n.t("authentication.cie.pin.lockedCiePinTitle"),
+        subtitle: I18n.t("authentication.cie.pin.lockedCiePinContent"),
+        action: createMessageAction({
+          label: I18n.t("global.buttons.close"),
+          onPress: navigateToAuthenticationScreen
+        }),
+        secondaryAction: createMessageAction({
+          label: I18n.t("authentication.cie.pin.lockedSecondaryActionLabel"),
+          onPress: didYouForgetPuk
+        })
+      }
+    }),
+    [
+      createMessageAction,
+      didYouForgetPin,
+      didYouForgetPuk,
+      navigateToAuthenticationScreen,
+      navigateToCiePinScreen
+    ]
+  );
+
+  // This should never happen,
+  // but it's a good practice to have a default message
+  // in case of unexpected values of `remainingCount`.
+  const defaultMessageThatShouldNeverHappen: Message = React.useMemo(
+    () => ({
+      pictogram: "attention",
+      title: I18n.t("global.genericError"),
+      subtitle: `${remainingCount}`,
+      action: createMessageAction({
+        label: I18n.t("global.buttons.retry"),
+        onPress: navigateToCiePinScreen
+      }),
+      secondaryAction: createMessageAction({
+        label: I18n.t("global.buttons.close"),
+        onPress: navigateToAuthenticationScreen
+      })
+    }),
+    [
+      createMessageAction,
+      navigateToAuthenticationScreen,
+      navigateToCiePinScreen,
+      remainingCount
+    ]
+  );
+
+  const getMessage = React.useCallback(
+    (key: number) =>
+      key in messages ? messages[key] : defaultMessageThatShouldNeverHappen,
+    [defaultMessageThatShouldNeverHappen, messages]
+  );
+
+  const { pictogram, title, subtitle, action, secondaryAction } =
+    getMessage(remainingCount);
+
+  return (
```

Comment:
A small detail, more a personal opinion, since the object returned from `getMessage` has the same keys required by `OperationResultScreenContent` it can be spread directly in the component.

Reply:
Addressed in https://github.com/pagopa/io-app/pull/5748/commits/8dd067ffc6fa345c09c972e3fcd3cc33871c4828 !